hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/handle/dispatch/ChildDispatchFactoryTest.java

```diff
@@ -185,13 +185,27 @@ void keyVerifierWithNullCallbackAndAuthorizingKeysAsExpected() {
     }
 
     @Test
-    void keyVerifierOnlySupportsKeyVerification() {
-        final var derivedVerifier = ChildDispatchFactory.getKeyVerifier(verifierCallback, DEFAULT_CONFIG, emptySet());
-        assertThatThrownBy(() -> derivedVerifier.verificationFor(Key.DEFAULT, assistant))
+    void keyVerifierWithCallbackAndAuthorizingKeysAsExpected() {
+        final var derivedVerifier =
+                ChildDispatchFactory.getKeyVerifier(a -> true, DEFAULT_CONFIG, Set.of(A_CONTRACT_ID_KEY));
+        assertThat(derivedVerifier.verificationFor(Key.DEFAULT).passed()).isTrue();
+        assertThat(derivedVerifier.verificationFor(Key.DEFAULT, (k, v) -> true).passed())
+                .isTrue();
+        assertThatThrownBy(() -> derivedVerifier.verificationFor(Bytes.EMPTY))
                 .isInstanceOf(UnsupportedOperationException.class);
+        assertThat(derivedVerifier.numSignaturesVerified()).isZero();
+        assertThat(derivedVerifier.authorizingSimpleKeys()).containsExactly(A_CONTRACT_ID_KEY);
+        assertThat(derivedVerifier.authorizingSimpleKeys()).isNotEmpty();
```

Comment:
FYI, `assertThat` has a "fluid" API, you can write

```java
        assertThat(derivedVerifier.authorizingSimpleKeys()).isNotEmpty().containsExactly(A_CONTRACT_ID_KEY);
```

(though in fact that's redundant in this particular case, don't need the `isNotEmpty()` here)
(in general should be `containsOnly` as even if this test had more then one element to be contained the order doesn't matter)
(just spitting out AssertJ stuff for posterity, don't even think about it.)